src/System.Text.Encodings.Web/src/System/Text/Encodings/Web/DefaultJavaScriptEncoderBasicLatin.cs

```diff
@@ -73,122 +73,260 @@ public override bool WillEncode(int unicodeScalar)
             return NeedsEscaping((char)unicodeScalar);
         }
 
-        [MethodImpl(MethodImplOptions.AggressiveInlining)]
         public override unsafe int FindFirstCharacterToEncode(char* text, int textLength)
         {
             if (text == null)
             {
                 throw new ArgumentNullException(nameof(text));
             }
 
+            Debug.Assert(textLength >= 0);
+
+            if (textLength == 0)
+            {
+                goto AllAllowed;
+            }
+
             int idx = 0;
+            short* ptr = (short*)text;
+            short* end = ptr + (uint)textLength;
 
 #if NETCOREAPP
-            if (Sse2.IsSupported)
+            if (Sse2.IsSupported && textLength >= Vector128<short>.Count)
+            {
+                goto VectorizedEntry;
+            }
+
+        Sequential:
+#endif
+            Debug.Assert(textLength > 0 && ptr < end);
+
+            do
+            {
+                Debug.Assert(text <= ptr && ptr < (text + textLength));
+
+                if (NeedsEscaping(*(char*)ptr))
+                {
+                    goto Return;
+                }
+
+                ptr++;
+                idx++;
+            }
+            while (ptr < end);
+
+        AllAllowed:
+            idx = -1;
+
+        Return:
+            return idx;
+
+#if NETCOREAPP
+        VectorizedEntry:
+            short* vectorizedEnd = end - 2 * Vector128<short>.Count;
+            int index;
+
+            while (ptr <= vectorizedEnd)
+            {
+                Debug.Assert(text <= ptr && ptr <= (text + textLength - 2 * Vector128<short>.Count));
+
+                // Load the next 16 characters, combine them to one byte vector
```

Comment:
Should probably have a comment clarifying what the `packsswb` instruction does: chars that don't cleanly convert to ASCII bytes will get converted to somewhere in the range `[ 0x7F .. 0xFF ]`, which the `NeedsEscaping` method will detect.

Reply:
Added the comment.